core/trino-main/src/main/java/io/trino/server/CoordinatorModule.java

```diff
@@ -228,6 +233,30 @@ protected void setup(Binder binder)
                 .annotatedWith(ForNoMemoryAwarePartitionMemoryEstimator.class)
                 .to(ExponentialGrowthPartitionMemoryEstimator.Factory.class).in(Scopes.SINGLETON);
 
+        // output data size estimator
+        binder.bind(OutputDataSizeEstimatorFactory.class)
+                .to(CompositeOutputDataSizeEstimator.Factory.class)
+                .in(Scopes.SINGLETON);
+        binder.bind(ByTaskProgressOutputDataSizeEstimator.Factory.class).in(Scopes.SINGLETON);
+        binder.bind(BySmallStageOutputDataSizeEstimator.Factory.class).in(Scopes.SINGLETON);
+        binder.bind(ByEagerParentOutputDataSizeEstimator.Factory.class).in(Scopes.SINGLETON);
+        // use provider method returning list to ensure ordering
+        install(new AbstractConfigurationAwareModule() {
```

Comment:
Make it into a `@Provides` on the containing class (CoordinatorModule). Should be no need for inner class.

Reply:
I know I can do that - but I think it is beneficial for readability to keep all of that together.